plugin/storage/factory.go

```diff
@@ -137,8 +137,16 @@ func (*Factory) getFactoryOfType(factoryType string) (storage.Factory, error) {
 // Initialize implements storage.Factory.
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.metricsFactory = metricsFactory
-	for _, factory := range f.factories {
-		if err := factory.Initialize(metricsFactory, logger); err != nil {
+	for name, factory := range f.factories {
+		if err := factory.Initialize(
+			metricsFactory.Namespace(metrics.NSOptions{
+				Name: "storage",
```

Comment:
What if we define this in the callers, similar to how you do it in v2?

Reply:
Actually, nvm, the meta-factory is the equivalent of v2 storage extension, so technically speaking it's already "in the caller".

Reply:
@yurishkuro how would we define the tags if we do this in the caller?